management-interface.go

```diff
@@ -116,6 +116,27 @@ func getDeviceName() string {
 	return strings.SplitN(name, ".", 2)[0]
 }
 
+// Return the serial number for the Raspberr Pi in the device.
+func getRaspberryPiSerialNumber() string {
+	if runtime.GOOS != "windows" {
+		// 'Nix.  Run /proc/cpuinfo command to get the info we want.
+		out, err := exec.Command("cat", "/proc/cpuinfo").Output()
+		if err != nil || len(out) == 0 {
+			return ""
+		}
+		// Extract the serial number. It's in the last row bar 1.  The last row of the output is blank.
+		rows := strings.Split(string(out), "\n")
+		lastRow := rows[len(rows)-2]
```

Comment:
Rather than assuming the position of this line (which could change at some point), why not loop over all the lines to find the one that starts with `SERIAL:`?

Reply:
Yes I had considered that.  With all of these commands, the output can change at any time and things may break.  In this case though, the 'protection' is fairly obvious so I will do this.